src/app/g3w-ol/controls/screenshotcontrol.js

```diff
@@ -27,12 +27,19 @@ proto.change = function(layers=[]){
  * same origin URL of current application in order to avoid
  * CORS issue while getting map image.
  * 
+ * Layers that don't have a source URL are excluded (eg. base layers)
+ * 
  * @param {array} layers
  * 
  * @returns {boolean}
  */
 proto.checkVisible = function(layers=[]) {
-  return undefined === layers.find((layer) => !sameOrigin(layer.getSource() && layer.getSource().url, location));
+  // get layer.getConfig().source instead of layer.getSource()
+  // because for BaseLayer instance src/app/core/layers/baselayers/baselayer.js,
+  // getSource() return ol.source instance and not source
+  // configuration object
+  return undefined === layers.find((layer) => layer.getConfig().source && layer.getConfig().source.url && !sameOrigin(layer.getConfig().source.url, location));
```

Comment:
Are we trying to exclude the base layers? 👉 https://github.com/g3w-suite/g3w-client/pull/426/commits/3ca0c248ffc1e8850de2a8a10284c44e8493fa78

Either way, it's more explicit when using the `layer.isBaseLayer()` function:

```suggestion
  return undefined === layers.find((layer) => !layer.isBaseLayer() && !sameOrigin(layer.getSource && layer.getSource().url, location))
```